src/modules/aws_vpc/mappers/network_acl.ts

```diff
@@ -0,0 +1,205 @@
+
+  getNetworkAcl = crudBuilderFormat<EC2, 'describeNetworkAcls', AwsNetworkAcl | undefined>(
+    'describeNetworkAcls',
+    aclId => ({ NetworkAclIds: [aclId] }),
+    res => res?.NetworkAcls?.pop(),
+  );
+
+  getNetworkAcls = paginateBuilder<EC2>(paginateDescribeNetworkAcls, 'NetworkAcls', undefined, undefined);
+
+  deleteNetworkAclEntry = crudBuilder<EC2, 'deleteNetworkAclEntry'>('deleteNetworkAclEntry', input => input);
+
+  deleteNetworkAcl = crudBuilder<EC2, 'deleteNetworkAcl'>('deleteNetworkAcl', networkAclId => ({
+    NetworkAclId: networkAclId,
+  }));
+
+  cloud: Crud<NetworkAcl> = new Crud({
+    create: async (es: NetworkAcl[], ctx: Context) => {
+      const out = [];
+
+      for (const e of es) {
+        const client = (await ctx.getAwsClient(e.region)) as AWS;
+
+        // we need to wait until vpc is created
+        if (!e.vpc?.vpcId) continue;
+
+        const input: CreateNetworkAclCommandInput = {
+          VpcId: e.vpc?.vpcId,
+        };
+        if (e.tags && Object.keys(e.tags).length) {
+          const tags: Tag[] = Object.keys(e.tags).map((k: string) => {
+            return {
+              Key: k,
+              Value: e.tags![k],
+            };
+          });
+          input.TagSpecifications = [
+            {
+              ResourceType: 'network-acl',
+              Tags: tags,
+            },
+          ];
+        }
+        const res = await this.createNetworkAcl(client.ec2client, input);
+        if (res) {
+          // now we need to add the entries
+          for (const entry of e.entries ?? []) {
+            await this.createNetworkAclEntry(client.ec2client, entry);
+          }
+        }
+
+        // re-read the network acl to get the updated results
+        const rawAcl = await this.getNetworkAcl(client.ec2client, res?.NetworkAclId);
+        if (rawAcl) {
+          const newAcl = await this.networkAclMapper(rawAcl, e.region, ctx);
+          if (newAcl) {
+            newAcl.id = e.id;
+            await this.db.update(newAcl, ctx);
+            out.push(newAcl);
+          }
+        }
+      }
+      return out;
+    },
+    read: async (ctx: Context, id?: string) => {
+      if (!!id) {
+        const { networkAclId, region } = this.idFields(id);
+        const client = (await ctx.getAwsClient(region)) as AWS;
+        const rawAcl = await this.getNetworkAcl(client.ec2client, networkAclId);
+        if (!rawAcl || rawAcl.IsDefault) return undefined; // cannot read default acls
+        return await this.networkAclMapper(rawAcl, region, ctx);
+      } else {
+        const out: NetworkAcl[] = [];
+        const enabledRegions = (await ctx.getEnabledAwsRegions()) as string[];
+        await Promise.all(
+          enabledRegions.map(async region => {
+            const client = (await ctx.getAwsClient(region)) as AWS;
+            for (const eg of await this.getNetworkAcls(client.ec2client)) {
+              if (eg && !eg.IsDefault) {
+                // skip default acls
```

Comment:
hmm, interesting. Why do we do this? no one can mess with them or it is to avoid issues like accidentally deleting them?